snmplib/asn1.c

```diff
@@ -507,11 +507,6 @@ asn_parse_int(u_char * data,
         (errpre, bufp, data, asn_length, *datalength))
         return NULL;
 
-    if ((size_t) asn_length > intsize || (int) asn_length == 0) {
-        _asn_length_err(errpre, (size_t) asn_length, intsize);
-        return NULL;
-    }
-
```

Comment:
So, the issue is there are some cases where the controller is returning a 5 byte integer.  I think it's an overflow condition they didn't check for when they have really big integers.

Other than that, the message is correct.  The correct encoding would be:
```
# length 4, 4 data bytes
0x04 0xFF 0xFF 0xFF 0xFE
```
the actual encoding is:
```
# length 5, 5(ish) data bytes
0x05 0x00 0xFF 0xFF 0xFF 0xFE
```
the rest of the code seems to "just work".  And I don't think there's a security issue here because there's a buffer check just above this line.
But, I'd appreciate some extra eyeballs on this.

Reply:
That seems reasonable, I think. The buffer check would seem to prevent any overflow.... Maybe I'm paranoid, why not leave the check in for the length == 0 case? Or, only allow `asn_length` to be 1 byte longer than intsize, but not unbounded

Reply:
Paranoia is a healthy survival trait.  I'm pretty sure we aren't hitting this block with any regular frequency since it just "drops" the packet.  But keeping the safeties enabled as much as possible is hard to argue against.  I'll test with `+1` and see if things are good.